### DISCUSSION TITLE
addEventListener, the third parameter

### DISCUSSION THREAD

**Discussion by ericxin1982** [alpinejs/alpine#4123] [1. Help]
right_drawer.addEventListener(
		  'click',
		  function (e) {
			e.stopPropagation()
			right_drawer.children[0].classList.value = 'bx bxs-user bx-sm'
			right_box.style.right = 0
			right_box.children[0].style.display = 'block'
		  },
                  true
		)

Hi,

I have a question, my friend helped me create a drawer component, and I try to convert to alpine template component, and this click event is equal to x-on click and pass an event target in, and how about the addeventlistener third parameter, I checked google, the default value is false, so I can ignore it, and how about true, how I should handle it within x-on.

Thanks
Eric Xin

**Comment by ekwoka** [comment]
It's capture.

which you can do with `.capture` but you almost definitely do not want to use that anyway.

Basically, when an event happens in the DOM, the event does not actually start it's life at that place (you click on element). It actually starts it's life at the top, the window/document, and it goes down the tree in the `capture` phase. Event listeners with `capture` as true will be given the event here. Calling `stopPropagation` will not stop the bubbling here.

then the event gets to the element you clicked on, it runs capture handlers there, and then run all the normal handlers there at that spot. and THEN if `stopPropagation` hasn't been called, it will start bubbling up triggering normal handlers.

You shouldn't use capture unless you have particularly clear and important reason to.

**Comment by ericxin1982** [comment]
Got it, thanks.
